app/src/main/java/com/infomaniak/mail/ui/main/thread/PrintMailFragment.kt

```diff
@@ -0,0 +1,100 @@
+import android.view.View
+import android.view.ViewGroup
+import android.webkit.WebView
+import androidx.core.view.get
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import androidx.navigation.fragment.navArgs
+import com.infomaniak.mail.R
+import com.infomaniak.mail.data.LocalSettings
+import com.infomaniak.mail.databinding.FragmentPrintMailBinding
+import dagger.hilt.android.AndroidEntryPoint
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class PrintMailFragment : Fragment() {
+
+    @Inject
+    lateinit var localSettings: LocalSettings
+
+    private var _binding: FragmentPrintMailBinding? = null
+    private val binding get() = _binding!! // This property is only valid between onCreateView and onDestroyView
+
+    private val navigationArgs: PrintMailFragmentArgs by navArgs()
+    private val threadViewModel: ThreadViewModel by viewModels()
+    private val threadAdapter inline get() = binding.messagesList.adapter as ThreadAdapter
+
+    private var subject = ""
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View {
+        return FragmentPrintMailBinding.inflate(inflater, container, false).also { _binding = it }.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?): Unit = with(threadViewModel) {
+        super.onViewCreated(view, savedInstanceState)
+        setupAdapter()
+        threadLive.observe(viewLifecycleOwner) { thread ->
+            thread?.subject?.let { subject = it }
+        }
+        messagesLive.observe(viewLifecycleOwner) { messages ->
+            threadAdapter.submitList(messages.filter { it.uid == navigationArgs.messageUid })
+        }
+        navigationArgs.openThreadUid?.let {
+            reassignThreadLive(it)
+            reassignMessagesLive(it)
+        }
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+
+    override fun onPause() {
+        super.onPause()
+        findNavController().popBackStack()
+    }
+
+    private fun setupAdapter() = with(binding.messagesList) {
+        adapter = ThreadAdapter(
+            shouldLoadDistantResources = true,
+            isForPrinting = true,
+            onBodyWebviewFinishedLoading = {
+                CoroutineScope(Dispatchers.Main).launch {
+                    val webViewPrintAdapter = getWebViewToPrint().createPrintDocumentAdapter(subject)
+                    val printManager = requireContext().getSystemService(Context.PRINT_SERVICE) as PrintManager
+                    printManager.print(subject, webViewPrintAdapter, null)
+                }
+            },
+        )
+    }
+
+    private fun getWebViewToPrint(): WebView = with(binding.messagesList[0]) { findViewById(R.id.bodyWebView) }
```

Comment:
Wouldn't we want the `fullMessageWebView` instead?

Reply:
`fullMessageWebView` doesn't have all messages of the thread, but it has the history of other emails if it was included, I think we want to also print this